filebeat/inputsource/network.go

```diff
@@ -43,4 +43,5 @@ type TLSMetadata struct {
 }
 
 // NetworkFunc defines callback executed when a new event is received from a network source.
+// data is a safe reference only for the lifespan of this func. Any offline usage should be done on a copy.
```

Comment:
:+1: 

I wonder why NetworkFunc is an alias, There is no need for the `=`. Can you remove that as well please?

Let's add details about lifetimes to SplitHandlerFactory (as well, or instead). In the end it is the HandlerFactory that governs the lifetime.